draft-ietf-ppm-dap.md

```diff
@@ -441,6 +441,7 @@ in the "type" field (within the DAP URN namespace
 | unrecognizedMessage        | The message type for a response was incorrect or the payload was malformed. |
 | unrecognizedTask           | An endpoint received a message with an unknown task ID. |
 | unrecognizedAggregationJob | An endpoint received a message with an unknown aggregation job ID. |
+| unrecognizedExtension      | An endpoint received a message with an unrecognized extension. |
```

Comment:
Do we need a new error type here, as well as the corresponding prescription elsewhere that an implementation MUST provide this error type in some scenario?

The question I think we should apply to all error types (including the existing ones) is whether any protocol participant actually behaves differently than they would if they just got some kind of generic "Bad Request" error.

Suppose a leader rejects a report because it contains an unrecognized extension. If the client sees a problem document with `unrecognizedExtension`, what is it supposed to do? Try again, trying different subsets of the extensions until it finds a combination the leader accepts? That doesn't make sense to me. If the client is configured to send some list of extensions, but the leader is configured to accept a different list of extensions, then I don't think the protocol can recover in-band. The client and leader operators would need to dig into why the configuration disagrees and take some action that is outside the scope of DAP.

That leads me to conclude that an error of type `unrecognizedMessage`, in response to which the client simply stops trying to upload that report, should suffice for the case of unrecognized extensions. The detail field of the problem document constructed by the server can contain detailed information on what's wrong with the message that aids operators in debugging, but I don't think the protocol needs to dictate a particular error code.

Reply:
I think it's better to reserve unrecognizedMessage for true parsing errors, but I take your point. There's no reason to spell out this extra error case because there's no automated response to it.

Reply:
I make the distinction between unrecognized extension and parsing error because it's possible to parse an unrecognized extension without knowing its structure. This is by virtue of the length tag.

Reply:
I maintain that for now, we should use `unrecognizedMessage`, and perhaps rename that error to just `badRequest` or something more generic. In particular I don't think this PR should add new error types.

Reply:
Done!